src/saveload/afterload.cpp

```diff
@@ -1455,6 +1455,7 @@ bool AfterLoadGame()
 
 		for (Vehicle *v : Vehicle::Iterate()) {
 			v->date_of_last_service += DAYS_TILL_ORIGINAL_BASE_YEAR;
+			v->date_of_last_service_newgrf = v->date_of_last_service;
```

Comment:
You don't actually need this line, as you will overwrite it later anyway. It also doesn't hurt. Up to you :)